discovery-provider/src/queries/query_helpers.py

```diff
@@ -328,6 +329,81 @@ def populate_track_metadata(session, track_ids, tracks, current_user_id):
     )
     save_count_dict = {track_id: save_count for (track_id, save_count) in save_counts}
 
+
```

Comment:
since we have track_ids and tracks here, i wonder if we could shortcut this logic if `remix_of` isn't defined. Not sure what would be more performant. Have you tested the implication off adding this here to the performance of the various endpoints?

Reply:
+1 this is a super heavyweight query and populate_track_metadata is used everywhere. if we could short circuit if it's not a remix that would cut down on runtime a lot in other queries as well

Reply:
Yup, added a check to only perform the query if there exists a track with the `remix_of` field and only query those track ids